Mention build-backend in regular/editable install differences (#11691)
pypa/pip: docs/html/topics/local-project-installs.md

With an editable install, you only need to perform a re-installation if you change the project metadata (eg: version, what scripts need to be generated etc). You will still need to run build commands when you need to perform a compilation for non-Python code in the project (eg: C extensions).

```{caution}

> It is possible to see behaviour differences such as differences in scope of installation between regular installs vs editable installs. In case you distribute the project as a "distribution package", users will see the behaviour of regular installs -- thus, it is important to ensure that regular installs work correctly.

Reviewer: ```suggestion
It is possible to see behaviour differences between regular installs vs editable installs. Precisely what those differences might be depends on the build backend, and you should check the backend documentation if the details matter to you. In case you distribute the project as a "distribution package", users will see the behaviour of regular installs -- thus, it is important to ensure that regular installs work correctly.
```

> It is possible to see behaviour differences between regular installs vs editable installs. Precisely what those differences might be depends on the build backend, and you should check the backend documentation if the details matter to you. In case you distribute the project as a "distribution package", users will see the behaviour of regular installs -- thus, it is important to ensure that regular installs work correctly.

Reviewer 2: _puts on copy editor hat_

```suggestion
It is possible to see behaviour differences between regular installs vs editable installs. These differences depend on the build-backend, and you should check the build-backend documentation for the details. In case you distribute the project as a "distribution package", users will see the behaviour of regular installs -- thus, it is important to ensure that regular installs work correctly.
```

We can be terse/more direct here.

- "Precisely what ... might be depends on ..." can be made terse/more direct.
- "check ... if the details matter" -> "check ... for details".
